src/parser/lexer/actions.rs

```diff
@@ -122,13 +122,11 @@ impl<S: LexemeSink> StateMachineActions for Lexer<S> {
 
     #[inline]
     fn create_start_tag(&mut self, _input: &[u8]) {
-        self.attr_buffer.borrow_mut().clear();
-
         self.current_tag_token = Some(StartTag {
             name: Range::default(),
             name_hash: LocalNameHash::new(),
             ns: Namespace::default(),
-            attributes: Rc::clone(&self.attr_buffer),
+            attributes: Vec::with_capacity(DEFAULT_ATTR_BUFFER_CAPACITY),
```

Comment:
I think we can just create the attributes here instead of reusing the buffer (which IIUC was an optimization).